server/compose/types/namespace.go

```diff
@@ -29,6 +29,28 @@ type (
 		Name string `json:"name"`
 	}
 
+	GlobalBlocks []GlobalBlock
+
+	GlobalBlock struct {
+		BlockID uint64 `json:"blockID,string,omitempty"`
+
+		Options map[string]interface{} `json:"options,omitempty"`
+		Style   PageBlockStyle         `json:"style,omitempty"`
+		Kind    string                 `json:"kind"`
+		XYWH    [4]int                 `json:"xywh"` // x,y,w,h
+		Meta    map[string]any         `json:"meta,omitempty"`
+
+		// Warning: value of this field is now handled via resource-translation facility
+		//          struct field is kept for the convenience for now since it allows us
+		//          easy encoding/decoding of the outgoing/incoming values
+		Title string `json:"title,omitempty"`
+
+		// Warning: value of this field is now handled via resource-translation facility
+		//          struct field is kept for the convenience for now since it allows us
+		//          easy encoding/decoding of the outgoing/incoming values
+		Description string `json:"description,omitempty"`
+	}
```

Comment:
Seems fine but you could consider just doing this:

```go
GlobalBlock struct {
		PageBlock
	}
```

Since you're basically doing a full blown copy of the page block this would be fine.
For page layouts, I preserve just the reference and coordinates and size since that's all that we care about there.